pkg/util/utils.go

```diff
@@ -244,3 +270,106 @@ func UpdateK8sResourceAnnotation(client client.Client, ctx *context.Context, k8s
 	}
 	return nil
 }
+
+// some resources has complex index, so set it type to string
+func GenerateID(res_id, prefix, suffix string, index string) string {
+	var id strings.Builder
+	if len(prefix) > 0 {
+		id.WriteString(prefix)
+		id.WriteString("_")
+	}
+
+	id.WriteString(res_id)
+	if len(index) > 0 {
+		id.WriteString("_")
+		id.WriteString(index)
+
+	}
+	if len(suffix) > 0 {
+		id.WriteString("_")
+		id.WriteString(suffix)
+	}
+	return id.String()
+}
+
+func GenerateDisplayName(res_name, prefix, suffix, project, cluster string) string {
+	var name strings.Builder
+	if len(prefix) > 0 {
+		name.WriteString(prefix)
+		name.WriteString("-")
+	}
+	if len(project) > 0 {
+		name.WriteString(project)
+		name.WriteString("-")
+	}
+	if len(cluster) > 0 {
+		name.WriteString(cluster)
+		name.WriteString("-")
+	}
+	name.WriteString(res_name)
+	if len(suffix) > 0 {
+		name.WriteString("-")
+		name.WriteString(suffix)
+	}
+	return name.String()
+}
+
+func GenerateTruncName(limit int, res_name, prefix, suffix, project, cluster string) string {
+	name := GenerateDisplayName(res_name, prefix, suffix, project, cluster)
+	return normalizeNamebyLimit(name, limit)
+}
+
+func BuildBasicTags(service common.Service, obj interface{}, namespaceID types.UID) []model.Tag {
```

Comment:
if we only need cluster, i suggest not to use the whole comon.service as param, only a "cluster string" should be enough

Reply:
+1, there is getCluster() func elsewhere.

Reply:
done